### ISSUE TITLE
Plugin silently fails with certain configurations

### ISSUE THREAD

**Issue by Venryx** [bfanger/eslint-plugin-only-warn#3]
Have been using the plugin with eslint v5 for months now, and it's worked great.

However, I recently updated to eslint v6, and the plugin is no longer working. (ie. all the eslint errors are showing up as errors again, in vscode)

I can provide further info if needed, but I don't think it would help; the plugin probably just doesn't match with the internals of eslint anymore.

(I may try to solve this one myself, mostly just posting it here for reference's sake.)

**Comment by Venryx** [comment]
I am using the same extension (https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint, latest version 1.9.1), and the same eslint version (v6.3.0), yet for me eslint is giving errors rather than warnings.

Here is a screenshot: https://i.imgur.com/xXFFBp9.png
![](https://i.imgur.com/xXFFBp9.png)

Notice that the search box is looking for messages with "error" matching the entry, and it shows results from eslint. This matches with the red underlining and the red icon next to it.

Since it's working for you, I suppose something must be wrong with my configuration -- despite it working just before I updated eslint.

If I figure out what was wrong, I'll post the solution back here.

**Comment by bfanger** [comment]
Could you share your eslint config? and a file with an eslint error?

**Comment by Venryx** [comment]
I don't know what changed, but after running `npm update`, the problem resolved.

Anyway, if hit again, I'll make a new issue since this one's title is misleading. (it's due to some other dependency, not eslint's update to v6)

**Comment by Venryx** [comment]
Still don't know all the details of what's causing the plugin to not (reliably) get applied, but I've found a way to solve it.

Debugging process:
1) First, I tried adding a `console.log` to the `enable()` function in this plugin's code. The log was not getting reached.
2) I knew from that that either the API for overriding `Linter.verify` had changed, or some other package was conflicting with the override.
3) I searched for other eslint plugins that override the `verify` function, to see if they had recently updated to adjust to some API change.
4) In doing so, I found this: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-unused-disable.md#known-limitations
5) Looking at [the function here](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/d6a9fd22220b3b3f6ca2c1b60f93b5d7f08f6493/lib/utils/patch.js#L139), I saw a different way to modify the `Linter.verify` function.
6) I tried copying that approach into the source code for `eslint-plugin-only-warn`, and it began working again!

Looking at the source code of the new approach, it seems that the reason it works is that it searches the webpack exports for any instance of `node_modules/eslint`, rather than only using the root package of it. This makes it work even when libraries have their own copies of `eslint`, instead of using the root project's.

Anyway, I've created a pull-request for this, as seen below.
